### PR TITLE
Fix new stories not appearing in the importFn

### LINKED ISSUE #354: [Bug] New story files are not injected with storyStoreV7
### What version of `vite` are you using?

2.9.1

### System info and storybook versions

```
System:
    OS: macOS 12.3.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Binaries:
    Node: 16.13.1 - ~/.volta/tools/image/node/16.13.1/bin/node
    Yarn: 1.22.17 - ~/.volta/bin/yarn
    npm: 8.3.0 - ~/.volta/tools/image/npm/8.3.0/bin/npm
  Browsers:
    Firefox: 99.0.1
    Safari: 15.4
  npmPackages:
    @storybook/addon-a11y: 6.5.0-alpha.49 => 6.5.0-alpha.49
    @storybook/addon-actions: 6.5.0-alpha.49 => 6.5.0-alpha.49
    @storybook/addon-docs: 6.5.0-alpha.49 => 6.5.0-alpha.49
    @storybook/addon-essentials: 6.5.0-alpha.49 => 6.5.0-alpha.49
    @storybook/addon-links: 6.5.0-alpha.49 => 6.5.0-alpha.49
    @storybook/addon-postcss: ^2.0.0 => 2.0.0
    @storybook/addons: 6.5.0-alpha.49 => 6.5.0-alpha.49
    @storybook/builder-vite: ^0.1.27 => 0.1.27
    @storybook/expect: ^27.5.2-0 => 27.5.2-0
    @storybook/jest: ^0.0.10 => 0.0.10
    @storybook/react: 6.5.0-alpha.49 => 6.5.0-alpha.49
    @storybook/test-runner: ^0.0.5-next.2 => 0.0.5-next.2
    @storybook/testing-library: ^0.0.9 => 0.0.9
    @storybook/testing-react: ^1.2.4 => 1.2.4
```

### Describe the Bug

When `storyStoreV7` is enabled, adding a new story file will cause the story to appear in the sidebar correctly, but trying to visit that story will cause an error, because the `importFn` for the new story file is not added.

### Link to Minimal Reproducible Example

_No response_

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by joshwooding**
Fixes #354

### CODE REVIEW

File: packages/builder-vite/code-generator-plugin.ts [hunk 1]
```diff
@@ -25,16 +25,21 @@ export function codeGeneratorPlugin(options: ExtendedOptions): Plugin {
...
         }
-        const storiesModule = moduleGraph.getModuleById(virtualStoriesFile);
+        const storiesModule = server.moduleGraph.getModuleById(virtualStoriesFile);
         if (storiesModule) {
           server.moduleGraph.invalidateModule(storiesModule);
         }
       });
+
+      server.watcher.on('all', (_e, path) => {
+        if (/\.stories\.([tj])sx?$/.test(path) || /.mdx$/.test(path)) {
```
Comment (IanVS) [inline packages/builder-vite/code-generator-plugin.ts:39]: should this be limited to `stories|story.mdx`?

Comment (joshwooding) [reply packages/builder-vite/code-generator-plugin.ts:39]: Any mdx could be shown I thought? Maybe we should re-use the stories regex though

Comment (IanVS) [reply packages/builder-vite/code-generator-plugin.ts:39]: I don't think anything other than a stories mdx will be adding new stories to the sidebar though, which I thought was the reason for this fix?

Comment (IanVS) [reply packages/builder-vite/code-generator-plugin.ts:39]: And yeah, I think using the stories regex would be best here, if it works.  I guess we wouldn't be able to if they're using the object-style configuration though, right?  Or is there a utility to turn that into a normal glob pattern?

Comment (joshwooding) [reply packages/builder-vite/code-generator-plugin.ts:39]: > I don't think anything other than a stories mdx will be adding new stories to the sidebar though, which I thought was the reason for this fix?

ah this just clicked. Good point

Comment (joshwooding) [reply packages/builder-vite/code-generator-plugin.ts]: I'll just stick with regex for now to keep it simple

File: packages/builder-vite/code-generator-plugin.ts [hunk 2]
```diff
@@ -25,16 +25,21 @@ export function codeGeneratorPlugin(options: ExtendedOptions): Plugin {
...
           server.moduleGraph.invalidateModule(appModule);
         }
-        const storiesModule = moduleGraph.getModuleById(virtualStoriesFile);
+        const storiesModule = server.moduleGraph.getModuleById(virtualStoriesFile);
         if (storiesModule) {
           server.moduleGraph.invalidateModule(storiesModule);
         }
       });
+
+      server.watcher.on('all', (_e, path) => {
```
Comment (IanVS) [inline packages/builder-vite/code-generator-plugin.ts:38]: Maybe a comment about why this is being added would be good to have, for anyone else coming across this code later?

**Review by IanVS — APPROVED** [review]

**Review by IanVS — APPROVED** [review]
:+1:

**Pull request merged**